ECMA2Yaml/ECMAHelper/ECMALoader.docs.cs

```diff
@@ -124,7 +126,7 @@ public Docs LoadDocs(XElement dElement, string filePath)
             return new Docs()
             {
                 Summary = NormalizeDocsElement(dElement.Element("summary")),
-                Remarks = remarksText,
+                Remarks = FormatTextIntoParagraphs(remarksText),
```

Comment:
Actually there are a lot of cases where writers use `<format type="text/markdown">` in `<remarks>` to tell ECMA2Yaml that they are using Markdown syntax, like this one: https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Windows/SystemColors.xml#L4504. So how about call `FormatTextIntoParagraphs()` inside `NormalizeDocsElement`, so that we only do this step when it's not Markdown?

When it comes to a format change, I usually diff the output yamls before/after my change in two source data sets: https://github.com/dotnet/dotnet-api-docs/tree/main/xml and https://github.com/Azure/azure-docs-sdk-dotnet/tree/master/xml.

Reply:
Thanks, @TianqiZhang.   That's helpful.   

There are definitely scenarios that I'm not accounting for here due to lack of awareness.   If you've got any kind of spec or documentation about formatting, please feel free to point me at that so I can take it into consideration.   To be transparent, I wasn't aware of  `<format>` -  it doesn't exist in the [C# Documentation Comments](https://docs.microsoft.com/dotnet/csharp/language-reference/language-specification/documentation-comments) section of the language spec nor in the [triple slash comments](https://dotnet.github.io/docfx/spec/triple_slash_comments_spec.html) section of the DocFX spec.

If you don't mind, I'd like to talk through things a bit to capture any corner cases.  In the base case, we know that this pattern is one we want to ignore and leave as-is:

```csharp
<remarks>
  <format type="text/markdown">
  <![CDATA[  

... MARKDOWN CONTENT ...
  
   ]]>
  </format>
</remarks>
```

What about a scenario where `<format>` is partial content within the `<remarks>` tag?  In the following example, if it is a valid case, I'd assume that we want to transform the first and third paragraphs and ignore the second due to `<format`> :

```csharp
<remarks>
  This is the first paragraph of text before we 
  drop down into some markdown formatting.

  <format type="text/markdown">
  <![CDATA[  

... MARKDOWN CONTENT ...
  
   ]]>
  </format>

  This is a trailing text block that represents a third
  paragraph that probably tells you wonderful things
</remarks>
```

The next consideration is an embedded `<format>`.  Given the following:

```csharp
<remarks>
  We're just here, minding our own business when a
  wild <format type="text/markdown"><![CDATA[  **_APPEARS_**  ]]></format>.  What now?

  This is a trailing text block that represents a third
  paragraph that probably tells you wonderful things.
</remarks>
```

I think that we'd want the transform to look like:
_(wrapping included here for readability, it would all be a single line in the actual transform)_

```csharp
<remarks>
  <p>We're just here, minding our own business when a  wild <format type="text/markdown"><![CDATA[  **_APPEARS_**  ]]></format>. What now?</p>
  <p>This is a trailing text block that represents a third paragraph that probably tells you wonderful things.</p>
</remarks>
```

Any other tags beyond `<format>` that we should account for, or other scenarios that may influence the transform?   I'd appreciate any insight that you may have.

Reply:
@jsquire thanks for the update, I'll diff the result by tomorrow and merge it if the result looks good in dotnet-api-docs repo.

Reply:
One more request @jsquire , sorry it took so long, I see there are a lot of diffs like below, which makes it difficult to review the real diffs. Is it possible to avoid touching the content if there's only 1 line?
![image](https://user-images.githubusercontent.com/5326582/118252800-bd32e980-b4db-11eb-8022-f4fc43179ea2.png)

Reply:
Hi @TianqiZhang.   Thanks for your feedback.  I adjusted parsing to no longer wrap a single line when no embedded paragraph tags are detected.  Tests and diff assets have been updated to reflect the new behavior.

I also  took a look through a few different scenarios for the MS Docs site output, and it looks like a single line of text gets rendered in `<p>` tags without explicitly adding them - so the user experience shouldn't be negatively impacted by this change.

Reply:
Diffs looking good, change merged, thanks again @jsquire !